WIP Avoid mutation during iteration bug in package object lazy open

retronym opened the pull request:
Still working on a test case...

lrytz commented on src/compiler/scala/tools/nsc/typechecker/Analyzer.scala:
```diff
@@ -80,8 +80,17 @@ trait Analyzer extends AnyRef
+        val toOpen = deferredOpen.toVector
+        for (sym <- toOpen) {
+          if (deferredOpen.remove(sym))
+            // CAREFUL -- this can mutate deferredOpen, hence the defensive code here!
```
can it add more symbols to `deferredOpen` that also need to be opened?

retronym replied:
AFAICT it only removes them

retronym replied:
https://github.com/scala/scala/blob/7e411f1dc176b7bf134ee596a5504418f63f5003/src/compiler/scala/tools/nsc/typechecker/Contexts.scala#L1890

retronym committed:
Fail fast if we mutate deferredOpen during iteration

Test for edge case with package objects load

Avoid mutation-during-iteration during package object initialization
